src/orion/algo/asha.py

```diff
@@ -194,18 +199,28 @@ def suggest(self, num=1):
                 return [candidate]
 
         if all(bracket.is_filled for bracket in self.brackets):
-            logger.debug('All brackets are filled.')
+            logger.warning('All brackets are filled.')
             return None
 
         for _attempt in range(100):
```

Comment:
This function is getting complicated. I would be in favor of extracting all sampling logic (from line to 202 to 223 in new code) into another method like we do in Hyperband. What do you think?